docs/source/app_developers_guide/environment_setup.rst

```diff
@@ -0,0 +1,207 @@
+
+
+Step One: Clone Repository
+--------------------------
+
+You'll need to have git installed in order to clone the Sawtooth Lake source
+code repository. You can find up-to-date installation instructions here:
+
+* `Git install instructions <https://git-scm.com/book/en/v2/Getting-Started-Installing-Git>`_
+
+.. note:: 
+
+  When checking out Sawtooth Lake on Windows for use with vagrant, you should
+  take steps to ensure that Windows-style CRLF line endings are not added to
+  the code. The bash scripts used by your vagrant VM will not run correctly 
+  with CRLF line endings. Git uses a configuration setting, *core.autocrlf*,
+  to control whether or not LF-style line endings are automatically converted
+  to CRLF-style line endings. `This setting should be set in such a way that 
+  CRLFs are not introduced into your repository 
+  <https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration>`_.
+
+Open up a terminal and run the following:
+
+.. code-block:: console
+
+   % cd $HOME
+   % mkdir project
+   % cd project
+   % git clone https://github.com/hyperledger/sawtooth-core.git
+
+.. note::
+
+  On a Windows environment, the suggested version of the last command
+  above is:
+
+  .. code-block:: console
+
+      C:\> git clone https://github.com/hyperledger/sawtooth-core.git
+      --config core.autocrlf=false
+
+
+Environment Startup
+-------------------
+
+To start up the environment, run:
+
+.. code-block:: console
+
+  % cd sawtooth-core
+  % docker-compose -f docker/compose/sawtooth-demo.yaml up
+
+Downloading the docker images that comprise the Sawtooth Lake demo
+environment can take serveral minutes. Once you see the containers
+registering and creating intial blocks you can move on to the next step.
+
+.. code-block:: console
+
+  Attaching to compose_validator_1, compose_tp_xo_python_1, compose_client_1, compose_tp_intkey_python_1, compose_tp_config_1, compose_rest_api_1
+  validator_1         | writing file: /etc/sawtooth/keys/validator.priv
+  validator_1         | writing file: /etc/sawtooth/keys/validator.pub
+  validator_1         | Generating /var/lib/sawtooth/genesis.batch
+  tp_xo_python_1      | [19:03:47 DEBUG   selector_events] Using selector: ZMQSelector
+  validator_1         | [19:03:47.537 INFO     path] Skipping path loading from non-existent config file: /etc/sawtooth/path.toml
+  tp_xo_python_1      | [19:03:47 INFO    core] register attempt: OK
+  validator_1         | [19:03:47.538 INFO     cli] config [path]: config_dir = "/etc/sawtooth"
+  tp_intkey_python_1  | [19:03:47 DEBUG   selector_events] Using selector: ZMQSelector
+  validator_1         | [19:03:47.538 INFO     cli] config [path]: key_dir = "/etc/sawtooth/keys"
+  tp_intkey_python_1  | [19:03:47 INFO    core] register attempt: OK
+
+Open a new terminal so we can connect to the client container:
+
+.. code-block:: console
+
+  % docker exec -it compose_client_1 bash
+
+Your environment is ready! Continue on to :doc:`intro_to_sawtooth`
```

Comment:
This is not true. Any work done in this environment will be lost once the container exits. The demo compose file as it stands is only useful for interacting with sawtooth temporarily. In order to use it for app development, some additional things would need to be added such as mounting a host directory into the container.